File: Phase_1/Day-6_04-08-2026/TargetSum.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> add(vector<int> &nums, int num) {
    vector<int> out;
    for (int i = 0; i < nums.size(); i++) {
        out.push_back(nums[i] + num);
        out.push_back(nums[i] - num);
    }
    return out;
}
int find(vector<int> &nums, int target) {
    int N = nums.size();
    vector<int> out = {0};
    int c = 0;
    for (int i = 0; i < N; i++) {
        out = add(out, nums[i]);
    }

    for (int i = 0; i < out.size(); i++) {
        if (out[i] == target) {
            c++;
        }
    }
    return c;

}

int main() {

}

File: Phase_1/Day-6_04-08-2026/TargetSum.java
import java.util.*;

public class TargetSum {
    private List<Integer> ad(List<Integer> v, int num) {
        List<Integer> out = new ArrayList<>();
        for (int val : v) {
            out.add(val + num);
            out.add(val - num);
        }
        return out;
    }
    public int findTargetSumWays(int[] nums, int target) {
        int N = nums.length;
        List<Integer> cur = new ArrayList<>();
        cur.add(0);

        for (int i = 0; i < N; i++) {
            cur = ad(cur, nums[i]);
        }

        int c = 0;
        for (int val : cur) {
            if (val == target) {
                c++;
            }
        }
        return c;
    }
}